### PR TITLE
fix(otel_batch_processor): don't divide `max_queue_size` by word-size

### PR DESCRIPTION
**Pull request by SergeTupchiy**
There is no need to convert `max_queue_size` to words, as it is compared with the number of objects in ETS table: https://github.com/open-telemetry/opentelemetry-erlang/blob/main/apps/opentelemetry/src/otel_batch_processor.erl#L278.

### CODE REVIEW

**Comment by bryannaegele** [comment]
I think that might actually be a bug in the code you linked as I think the intention is to limit memory consumption, not the number of spans. The rules around limits are focused around things that affect memory pressure.

I think the info check should actually be this:

`case ets:info(?CURRENT_TABLE(RegName), memory) of`

**Comment by SergeTupchiy** [comment]
> I think that might actually be a bug in the code you linked as I think the intention is to limit memory consumption, not the number of spans. The rules around limits are focused around things that affect memory pressure.
> 
> I think the info check should actually be this:
> 
> `case ets:info(?CURRENT_TABLE(RegName), memory) of`

Hi @bryannaegele, 
Thanks for your reply. 
It used to be compared against memory size (in words) indeed: 
- https://github.com/open-telemetry/opentelemetry-erlang/commit/571d4af0b3c61ce29843d6edc4f09260e9b097d8#diff-5bd137db87b7d675fde66547dccf5b47aadebcd8f6f1401bc5d5a6e2ff3ebc7bR172
- https://github.com/open-telemetry/opentelemetry-erlang/commit/571d4af0b3c61ce29843d6edc4f09260e9b097d8#diff-f94a0d999fc7707f9c724767c27af9e5009313ecdcb865b0da2b5f364c45cec6L167

I'd be happy to fix it to use memory instead, but I think we also need to account for some pre allocated memory per table:
```
> ets:info('otel_batch_processor_otel_export_table1_<0.4149.0>', memory).
1377
> ets:info('otel_batch_processor_otel_export_table1_<0.4149.0>', size).  
0
```
Basically, an empty table already occupies more words than the default max_queue_size: `2048 div erlang:system_info(wordsize) = 256`

**Comment by tsloughter** [comment]
Wow, yea, this got all messed up...

Hm, because spans have options for limits on attributes maybe we should just use the # of spans instead of size of the table?

**Comment by bryannaegele** [comment]
What if we supported both? `integer` for number of entries, `{memory_unit, integer}` for memory size? I suspect most folks would want to use a memory cap as span sizes and volume can vary so greatly and should be the default method.

**Comment by SergeTupchiy** [comment]
@tsloughter , @bryannaegele,
Found another related issue: check_table_size timeout has not been started at all, attempted to fix it in this commit: https://github.com/open-telemetry/opentelemetry-erlang/pull/635/commits/42b8376c09b31491c42519e1ba8e54b1a827eba2

**Comment by tsloughter** [comment]
I'm thinking we should go with this and can add a `max_queue_size_bytes` if desired in the future if desired.

@bryannaegele does that sound good?

**Review by tsloughter — APPROVED** [review]

**Comment by bryannaegele** [comment]
Yep. I'm fine with length

**Review by bryannaegele — APPROVED** [review]
